x/nexus/keeper/lockable_asset.go

```diff
@@ -190,6 +188,18 @@ func getCoinType(ctx sdk.Context, nexus types.Nexus, denom string) (types.CoinTy
 	}
 }
 
+// isFromExternalCosmosChain returns true if the asset origins from cosmos chains
+func isFromExternalCosmosChain(ctx sdk.Context, nexus types.Nexus, ibc types.IBCKeeper, denom string) bool {
+	chain, ok := nexus.GetChainByNativeAsset(ctx, denom)
```

Comment:
IBC coins aren't registered by their ICS20 naming, `ibc/{hash}`, but by the base denom of their fully qualified path, `transfer/channel-x/uosmo` gets registered as `uosmo`

Reply:
We know. Denom here is indeed the base denom.